### PR TITLE
29 hyperparameter search

### PR DESCRIPTION
**Pull request by ferbcastro**
Vejam o `train.py`, `grid_search.py` e `loss.py`. Sei que loss deveria estar em outra branch mas fiz aqui msm.

### CODE REVIEW

File: src/hyperparameters/grid_search.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+print('Loading datasets...')
+train, test = ml.setup(TRAIN_SOURCE, TEST_SOURCE, cuda_src=CUDA_SRC)
+print('Datasets loaded')
+
+best_acc = -1
+
+for params in permute(param_grid):
+    b = params['epochs']
+    ep = params['batches']
+    ec = params['encoding']
```
Comment (d4vi13) [inline src/hyperparameters/grid_search.py]: aqui tem um problema que a estrurura da rede ta fixa ne, entao no final ele sempre ta dropando de 2450 pra ec. 

O que to pensando eh fazer o seguinte da pra adicionar aqui um compression_rate pra ser permutado tbm, dai daria pra passar pro construtor esse compression_rate e dava pra construir as camadas assim. 

Dai tipo se tu passa tipo 0,5 de compression rate ele vai caindo pela metade toda vez ate chegar no encoding size

O que acha?? 

posso fazer

File: src/training/train.py [hunk 2]
```diff
@@ -17,42 +15,39 @@
...
-    print("Usage: python3 train.py path_to_csv [batch_size]")
-    print("  path_to_csv: Path to the CSV file containing the data")
-    print("  batch_size: Optional batch size for training (default: 32)")
-    sys.exit(1)
+from models.autoencoder import UndercompleteAE
+import training.loss as loss
 
-CSV_SOURCE = sys.argv[1]
-BATCH_SIZE = int(sys.argv[2]) if len(sys.argv) > 2 else 32
+CUDA_SRC = 0
```
Comment (d4vi13) [inline src/training/train.py]: q isso?

File: src/training/train.py [hunk 3]
```diff
@@ -99,12 +96,23 @@ def train(X: pd.DataFrame, batch_size: int = 32):
...
+    X_tensor = X_tensor.to(device)
+
     with torch.no_grad():
-        encoded_data = model.encoder(X_tensor).detach().numpy()
+        output = model(X_tensor)
+        acc = (output.round() == X).float().mean()
+
+    return acc
 
-    return encoded_data, model
```
Comment (d4vi13) [inline src/training/train.py]: Nao entendi como que a gente vai pegar os dados encodados dai
